keystores/src/main/java/com/palantir/conjure/java/config/ssl/SslSocketFactories.java

```diff
@@ -210,6 +210,20 @@ public static TrustContext createTrustContext(SslConfiguration config) {
         return TrustContext.of(context.getSocketFactory(), getX509TrustManager(trustManagers));
     }
 
+    /**
+     * Create an {@link TrustContext} from the provided configuration and provider. See {@link #createX509TrustManager}
+     * and {@link #createSslSocketFactory}.
+     *
+     * @param config an {@link SslConfiguration} describing the trust store and key store configuration
+     * @param provider The preferred security {@link Provider}
+     */
+    public static TrustContext createTrustContext(SslConfiguration config, Provider provider) {
+        TrustManager[] trustManagers = createTrustManagers(config);
+        KeyManager[] keyManagers = createKeyManagers(config);
+        SSLContext context = createSslContext(trustManagers, keyManagers, provider);
+        return TrustContext.of(context.getSocketFactory(), getX509TrustManager(trustManagers));
+    }
```

Comment:
I suspect we won't actually use the new overload methods, let's undo this part for this PR

Reply:
handled in dc3995258d697797736c97f45a18a24c2da4fae8